back/src/Services/SocketManager.ts

```diff
@@ -381,23 +380,11 @@ export class SocketManager {
             response.setRoomid(roomId);
 
             // Check to see if there's an active meeting for this room set up already that we don't know about yet
-            // ToDo use const for https://webexapis.com/v1
-            // ToDo make sure using inProgress is save enough (https://developer.webex.com/docs/api/v1/meetings/list-meetings)
-            const params = `meetingType=meeting&state=inProgress&siteUrl=${WEBEX_SITE_URL}`;
-            const res = await Axios.get(`https://webexapis.com/v1/meetings?integrationTag=${roomId}&${params}`, {
-                headers: {
-                    "Content-Type": "application/json",
-                    Authorization: `Bearer ${accessToken}`,
-                },
-            });
-            // ToDo add error handling
-            console.log("[Back] Looking up meeting, got: ", res.data);
-            const legalMeets = res?.data?.items;
-            console.log("[Back] Legal meetings to choose from: ", legalMeets);
-            let meetingId = legalMeets && legalMeets[0]?.id;
+            const meet = this.webexMeetings.get(roomId);
+            let meetingId = meet?.meet.id;
             // todo get meetingLink by meetingId #11
-            let meetingLink;
-            if (!meetingId) {
+            let meetingLink = meet?.meet.sipAddress;
+            if (!meetingId || (meet && Date.parse(meet.meet.end) <= Date.now()) || !meetingLink) {
```

Comment:
Why do we need `!meetingId || (meet && Date.parse(meet.meet.end) <= Date.now()) ||` part?
I would propose just `if (!meetingLink)`.
The check if the meeting is ended already could be guaranteed by deleting the meeting entity from temporary store or later from database every time the meeting ended

Reply:
That would make sense, I'm just worried about the case when all users in a meeting close the browser tab before the meeting ends :) In doing so, the backend wouldn't know to delete the meeting, right?

Reply:
could we use `onDestroy` for this case?

`Date.parse(meet.meet.end) <= Date.now())` will not solve the problem. For example: `meet.meet.end` is still in the future, but everybody left the meeting.